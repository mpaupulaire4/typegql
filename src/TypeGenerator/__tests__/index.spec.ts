// import { PubSub } from 'graphql-subscriptions';
import {
  Prop,
  Type,
  Query,
  Subscription,
  Mutation,
} from '../../'
import {
  GenerateTypes
} from '../'

@Type()
export class User {
  @Prop('Int') id: number;
  @Prop() name: string;
  @Prop('Int') number: number;
}

export class UserContoller {
  @Query({
    type: '[User!]!'
  })
  query(){}

  @Mutation({
    type: 'String',
  })
  mutation(){}

  @Subscription({
    type: 'String'
  })
  subscription(){}
}

describe('Type Generator', () => {
  it('should generate type strings from decorators', () => {
    expect(GenerateTypes()).toMatchSnapshot()
  })

  it('extend: true  should extend Query, Mutation, Subscription', () => {
    expect(GenerateTypes({
      extend: true
    })).toMatchSnapshot()
  })

  it('should extend Query and User', () => {
    expect(GenerateTypes({
      extend: {
        Query: true,
        User: true
      }
    })).toMatchSnapshot()
  })

  it('should exclude Query', () => {
    expect(GenerateTypes({
      generateQueries: false
    })).toMatchSnapshot()
  })

  it('should exclude Mutations', () => {
    expect(GenerateTypes({
      generateMutations: false
    })).toMatchSnapshot()
  })

  it('should exclude Subscriptions', () => {
    expect(GenerateTypes({
      generateSubscriptions: false
    })).toMatchSnapshot()
  })

  it('should only include root types', () => {
    expect(GenerateTypes({
      generateTypes: false
    })).toMatchSnapshot()
  })

  it('should include schema files', () => {
    expect(GenerateTypes({
      schemas: [
        '**/*.type-generator.gql'
      ]
    })).toMatchSnapshot()
  })

})

