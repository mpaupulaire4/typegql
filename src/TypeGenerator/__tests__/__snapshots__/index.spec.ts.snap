// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Generator extend: true  should extend Query, Mutation, Subscription 1`] = `
"type User {
  id: Int
  name: String
  number: Int
}
extend type Query {
  query: [User!]!
}
extend type Mutation {
  mutation: String
}
extend type Subscription {
  subscription: String
}"
`;

exports[`Type Generator should exclude Mutations 1`] = `
"type User {
  id: Int
  name: String
  number: Int
}
type Query {
  query: [User!]!
}
type Subscription {
  subscription: String
}"
`;

exports[`Type Generator should exclude Query 1`] = `
"type User {
  id: Int
  name: String
  number: Int
}
type Mutation {
  mutation: String
}
type Subscription {
  subscription: String
}"
`;

exports[`Type Generator should exclude Subscriptions 1`] = `
"type User {
  id: Int
  name: String
  number: Int
}
type Query {
  query: [User!]!
}
type Mutation {
  mutation: String
}"
`;

exports[`Type Generator should extend Query and User 1`] = `
"extend type User {
  id: Int
  name: String
  number: Int
}
extend type Query {
  query: [User!]!
}
type Mutation {
  mutation: String
}
type Subscription {
  subscription: String
}"
`;

exports[`Type Generator should generate type strings from decorators 1`] = `
"type User {
  id: Int
  name: String
  number: Int
}
type Query {
  query: [User!]!
}
type Mutation {
  mutation: String
}
type Subscription {
  subscription: String
}"
`;

exports[`Type Generator should include schema files 1`] = `
"type User {
  id: Int
  name: String
  number: Int
}
type Query {
  query: [User!]!
}
type Mutation {
  mutation: String
}
type Subscription {
  subscription: String
}
input TestInput {
  test: Int!
}
input TestInput2 {
  test: Int!
}"
`;

exports[`Type Generator should only include root types 1`] = `
"type Query {
  query: [User!]!
}
type Mutation {
  mutation: String
}
type Subscription {
  subscription: String
}"
`;
